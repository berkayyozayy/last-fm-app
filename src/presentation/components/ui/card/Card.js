import styles from "./Card.module.css";

function Card({ image, label, count, listeners }) {
  return (
    <div className={styles["card"]}>
      <div className={styles["image"]}>
        <img src={image} alt={label} />
      </div>
      <div className={styles["content"]}>
        <div className={styles["info"]}>
          <p className={styles["title"]}>User</p>
          <p>{label}</p>
        </div>
        <div className={styles["sub-info"]}>
          <p>Playcount: {count}</p>
          <p>Listeners: {listeners}</p>
        </div>
      </div>
    </div>
  );
}

export default Card;
