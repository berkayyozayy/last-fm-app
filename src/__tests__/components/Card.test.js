import { render } from "@testing-library/react";
import Card from "../../presentation/components/ui/card/Card";

describe("card component", () => {
  test("it should render", () => {
    const { container } = render(<Card />);
    expect(container).toMatchSnapshot();
  });
});
