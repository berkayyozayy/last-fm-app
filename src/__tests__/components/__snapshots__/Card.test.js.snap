// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card component it should render 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="image"
    >
      <img />
    </div>
    <div
      class="content"
    >
      <div
        class="info"
      >
        <p
          class="title"
        />
        <p
          class="label"
        />
      </div>
      <div
        class="sub-info"
      >
        <p>
          Playcount: 
        </p>
        <p>
          Listeners: 
        </p>
      </div>
    </div>
  </div>
</div>
`;
